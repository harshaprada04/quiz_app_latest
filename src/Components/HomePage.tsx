import { useNavigate } from "react-router";
import React, { useState, ChangeEvent } from "react";
import { useContext } from "react";
import Context from "../Contexts/contexts";
import classes from "./homepage.module.css";
import { HomePageDetails } from "../Interface/interface";
import {
  FormControl,
  TextField,
  InputLabel,
  Select,
  MenuItem,
  Container,
  Button,
} from "@mui/material";


function HomePage() {
  const [userDetails, setUserDetails] = useState<HomePageDetails>({
    firstName: "",
    lastName: "",
    gender: "",
    prefferedLanguage: "",
  });
  const navigate = useNavigate();
  const context = useContext(Context);

  function clickHandler() {
    navigate("/question/:questionId");
    if (userDetails.prefferedLanguage.length > 0) {
      return context.setSelectedLanguage(userDetails.prefferedLanguage);
    }
  }
  return (
      <div className={classes.dispaly}>
        <header>Login</header>
        <Container >
          <div className={classes.allignment}>
            <TextField
            name= "firstname"
            sx={{ minWidth: 220 }}
              id="outlined-basic"
              label="First Name"
              variant="outlined"
              onChange={(e) =>
                setUserDetails((prev) => ({
                  ...prev,
                  firstName: e.target.value,
                }))
              }
              value={userDetails.firstName}
              autoComplete="off"
            />
            <TextField
            sx={{ minWidth: 220 }}
              id="outlined-basic"
              label="Last Name"
              variant="outlined"
              onChange={(e) =>
                setUserDetails((prev) => ({
                  ...prev,
                  lastName: e.target.value,
                }))
              }
              value={userDetails.lastName}
              autoComplete="off"
            />

            <FormControl sx={{ minWidth: 220 }}>
              <InputLabel id="demo-simple-select-label">Gender</InputLabel>
              <Select
              inputProps={{ "data-testid": "gender-selection" }}
                labelId="demo-simple-select-label"
                id="demo-simple-select"
                value={userDetails.gender}
                label="Gender"
                onChange={(e) =>
                  setUserDetails((prev) => ({
                    ...prev,
                    gender: e.target.value,
                  }))
                }
              >
                <MenuItem value="Male">Male</MenuItem>
                <MenuItem value="Female">Female</MenuItem>
                <MenuItem value="Others">Others</MenuItem>
              </Select>
            </FormControl>

            <FormControl sx={{ minWidth: 220 }}>
              <InputLabel id="demo-simple-select-label">Language</InputLabel>
              <Select
              inputProps={{ "data-testid": "language-selection" }}
                labelId="demo-simple-select-label"
                id="demo-simple-select"
                value={userDetails.prefferedLanguage}
                label="Language"
                onChange={(e) =>
                  setUserDetails((prev) => ({
                    ...prev,
                    prefferedLanguage: e.target.value,
                  }))
                }
              >
                <MenuItem  value="English">English</MenuItem>
                <MenuItem value="Hindi">Hindi</MenuItem>
              </Select>
            </FormControl>
            </div>
        </Container>
        <Button
          variant="contained"
          color="primary"
          disabled={
            !(
              userDetails.prefferedLanguage &&
              userDetails.firstName &&
              userDetails.lastName
            )
          }
          className={classes.btn}
          name="submit"
          type="submit"
          onClick={clickHandler}
        >
          Submit
        </Button>
      </div>
  );
}

export default HomePage;
