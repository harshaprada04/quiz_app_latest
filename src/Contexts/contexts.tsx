import { createContext, useState } from "react";
import { Details } from "../Interface/interface";

const Context = createContext({
  questionList: [],
  setQuestionList: (a: Details[]) => {},
  selectedLanguage: "",
  setSelectedLanguage: (a: {}) => {},
  count: { correctAns: 0, wrongAns: 0 },
  setCount: () => {},
  getQuestionById: (id: number): any => {},
  changeHandler: (id: number, answer: string): any => {}
});

export function ContextProvider(props: any) {
  const [questionList, setQuestionList] = useState<Details[]>([]);
  const [selectedLanguage, setSelectedLanguage] = useState<string>("");
  const [count, setCount] = useState({ correctAns: 0, wrongAns: 0 });

  function changeHandler(id: number, answer: string) {
    setQuestionList(
      questionList.map((question) => {
        if (question.id === id) {
          question.isAnswered = answer.length ? true : false;
          question.selectedAnswer = answer;
        }
        return question;
      })
    );
    if (id == questionList.length) {
      submitHandler(questionList);
    }
  }

  function submitHandler(questionList: any) {
    setCount({ correctAns: 0, wrongAns: 0 });
    for (let i: number = 0; i < questionList.length; i++) {
      if (questionList[i].isAnswered === true) {
        let isCorrectAns = false;
        if (questionList[i].actualAnswer instanceof Array) {
          isCorrectAns = questionList[i].selectedAnswer.every(
            (selected: string) =>
              questionList[i].actualAnswer.some(
                (actual: string) => actual === selected
              )
          );
        } else {
          isCorrectAns =
            questionList[i].actualAnswer === questionList[i].selectedAnswer;
        }
        if (isCorrectAns) {
          setCount((prevState) => ({
            ...prevState,
            correctAns: prevState.correctAns + 1,
          }));
        } else {
          setCount((prevState) => ({
            ...prevState,
            wrongAns: prevState.wrongAns + 1,
          }));
        }
      }
    }
  }

  function getQuestionById(id: number) {
    return questionList.find((question: any) => question.id === id) || [];
  }


 
  const context: any = {
    questionList,
    count,
    setCount,
    selectedLanguage,
    setSelectedLanguage,
    changeHandler,
    setQuestionList,
    getQuestionById
  };

  return (
    <div>
      <Context.Provider value={context}>{props.children}</Context.Provider>
    </div>
  );
}

export default Context;
