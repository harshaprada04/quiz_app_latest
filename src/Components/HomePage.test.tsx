import { fireEvent, getByText, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter } from "react-router-dom";

import { act } from "react-dom/test-utils";



import HomePage from "./HomePage";


import { ContextProvider } from "../Contexts/contexts";


let AddRouting = () => {
  return (
    <BrowserRouter>
      <HomePage />
    </BrowserRouter>
  );
};

describe("Home", () => {
	it("renders", () => {
		render(<AddRouting />);
	});

	it("matches snapshot", () => {
		const view = render(<AddRouting />);
		expect(view).toMatchSnapshot();
	});
})
it("Button is disabled or not", async () => {
  render(<AddRouting />);
  expect(
    await screen.getByRole("button", {
      name: /submit/i,
    })
  ).toBeDisabled();
});

// it("Button has click event or not", async () => {
//   render(<AddRouting />);
//   const button = await screen.getByRole("button", {name: /submit/i,})
//   const click = fireEvent.click(button)
//   expect(click).toBe(true);
// });

// test("Form completion", async () => {
//   render(<AddRouting/>)
//   let firstname = await screen.getByLabelText("First Name");
//   fireEvent.change(firstname, {target:{value:"Harsha"}})
//   expect(firstname).toHaveValue("Harsha")
//   let lastname = await screen.getByLabelText("Last Name");
//   fireEvent.change(lastname, {target:{value:"Adiga"}})
//   expect(lastname).toHaveValue("Adiga")
//   let gender = await screen.getByTestId("gender-selection");
//   let click = fireEvent.click(gender)
//   expect(click).toBe(true)
  
//   // let option = await screen.getByText("Male")
//   // fireEvent.click(option, {checked:true})
//   // expect(gender).toHaveValue("Male")
//   // fireEvent.click(gender)
//   // fireEvent.change(gender, {target:{value:"Male"}})
//   let language = await screen.getByTestId("language-selection" );
//   fireEvent.click(language)
 
//   let button = await screen.getByRole("button",{name:/submit/i})
//   expect(button).not.toBeDisabled()
// });
