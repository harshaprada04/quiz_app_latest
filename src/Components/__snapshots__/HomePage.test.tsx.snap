// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dispaly"
      >
        <header>
          Login
        </header>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="allignment"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1yd6u5z-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                First Name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-basic"
                  name="firstname"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      First Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1yd6u5z-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="outlined-basic"
                id="outlined-basic-label"
              >
                Last Name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="outlined-basic"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Last Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-1qot2op-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                id="demo-simple-select-label"
              >
                Gender
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="demo-simple-select-label demo-simple-select"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="demo-simple-select"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  data-testid="gender-selection"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Gender
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-1qot2op-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                id="demo-simple-select-label"
              >
                Language
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="demo-simple-select-label demo-simple-select"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="demo-simple-select"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  data-testid="language-selection"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Language
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled btn css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled=""
          name="submit"
          tabindex="-1"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dispaly"
    >
      <header>
        Login
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="allignment"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1yd6u5z-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              First Name
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="firstname"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    First Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1yd6u5z-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Last Name
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Last Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1qot2op-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="demo-simple-select-label"
            >
              Gender
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                data-testid="gender-selection"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Gender
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1qot2op-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="demo-simple-select-label"
            >
              Language
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                data-testid="language-selection"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Language
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled btn css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        name="submit"
        tabindex="-1"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
